server/src/github.com/alexeyknyshev/cpsrv/cpsrv_patch_test.go

```diff
@@ -0,0 +1,425 @@
+package main
+
+import (
+	//"bytes"
+	"encoding/json"
+	//"errors"
+	"fmt"
+	//"github.com/alexeyknyshev/gojsondiff"
+	//"github.com/alexeyknyshev/gojsondiff/formatter"
+	//"github.com/gorilla/mux"
+	//"io/ioutil"
+	//"log"
+	//"net/http"
+	//"net/http/httptest"
+	//"sort"
+	//"reflect"
+	"strconv"
+	"testing"
+)
+
+type PatchRequest struct {
+	Id             uint32   `json:"id,omitempty"`
+	Longitude      float64  `json:"longitude,omitempty"`
+	Latitude       float64  `json:"latitude,omitempty"`
+	Type           string   `json:"type,omitempty"`
+	BankId         uint32   `json:"bank_id,omitempty"`
+	TownId         uint32   `json:"town_id,omitempty"`
+	Address        string   `json:"address,omitempty"`
+	AddressComment string   `json:"address_comment,omitempty"`
+	MetroName      string   `json:"metro_name,omitempty"`
+	FreeAccess     *bool    `json:"free_access,omitempty"`
+	MainOffice     *bool    `json:"main_office,omitempty"`
+	WithoutWeekend *bool    `json:"without_weekend,omitempty"`
+	RoundTheClock  *bool    `json:"round_the_clock,omitempty"`
+	WorksAsShop    *bool    `json:"works_as_shop,omitempty"`
+	Schedule       Schedule `json:"schedule,omitempty"`
+	Tel            string   `json:"tel,omitempty"`
+	Additional     string   `json:"additional,omitempty"`
+	Rub            *bool    `json:"rub,omitempty"`
+	Usd            *bool    `json:"usd,omitempty"`
+	Eur            *bool    `json:"eur,omitempty"`
+	CashIn         *bool    `json:"cash_in,omitempty"`
+}
+
+type TestPatchReq struct {
+	Data   PatchRequest `json:"data"`
+	UserId uint         `json:"user_id"`
+}
+
+func getPatchExampleNewCP() *PatchRequest {
+	False := false
+	True := true
+	patchReq := PatchRequest{
+		Longitude:      37.6878262,
+		Latitude:       55.6946643,
+		Type:           "atm",
+		BankId:         322,
+		TownId:         4,
+		Address:        "г. Москва, Район Моей Мечты",
+		AddressComment: "ОАО UnderButtom",
+		MetroName:      "",
+		FreeAccess:     &True,
+		MainOffice:     &False,
+		WithoutWeekend: &False,
+		RoundTheClock:  &False,
+		WorksAsShop:    &True,
+		Schedule:       Schedule{},
+		Tel:            "",
+		Additional:     "",
+		Rub:            &True,
+		Usd:            &False,
+		Eur:            &False,
+		CashIn:         &False,
+	}
+	return &patchReq
+
+}
+
+func getPatchExampleExistCP() (*PatchRequest, string) {
+	patchReq := PatchRequest{
+		Id:     58552,
+		BankId: 2764,
+	}
+	exampleJson := "{\"schedule\":{},\"bank_id\":" + strconv.FormatUint(uint64(patchReq.BankId), 10) + "}"
```

Comment:
BTW, you can use [anonymous structs](https://talks.golang.org/2012/10things.slide#2)
Example: https://play.golang.org/p/WSsay8kpgo